internal/serve/httphandler/verifiy_receiver_registration_handler.go

```diff
@@ -112,6 +112,25 @@ func (v VerifyReceiverRegistrationHandler) processReceiverVerificationPII(
 	now := time.Now()
 	truncatedPhoneNumber := utils.TruncateString(receiver.PhoneNumber, 3)
 
+	// STEP 0: if customer-id exists in sep24 claims, use it to look up reciever, hash Phone Number  in db, compare against hashed mobile_number in sep24 claims
+	if receiverRegistrationRequest.CustomerID != "" {
```

Comment:
👋  The trigger to validate the customerID should not depend on it being in the request, it should instead depend on the `use-external-id` CLI configuration.

Otherwise the externalID check will be ignored if the externalID is not provided.

This suggestion shows how to wire this configuration: https://github.com/stellar/stellar-disbursement-platform-backend/pull/256#discussion_r1563061427